### DISCUSSION TITLE
Top speed gathering

### DISCUSSION THREAD

**Discussion by Lucasjnt** [theOehrly/Fast-F1#637] [General]
Hi I'm having issues to collect top speed , I'm seeing top speeds but there are bigger speeds that are not being found 

@register_analysis('Maior Velocidade por Equipe')
def generate_top_speed_by_team_plot(session):
    # Cria a figura e os eixos, usando o padrão do set_plot_style
    fig, ax = plt.subplots()

    # Aplica o estilo do gráfico usando set_plot_style
    set_plot_style(ax, session, "Maior Velocidade na Volta Mais Rápida por Equipe")

    # Filtra as voltas com valores válidos de velocidade e tempo de volta
    valid_laps = session.laps.dropna(subset=['SpeedST', 'LapTime'])

    # Encontra a volta mais rápida de cada equipe com base no menor LapTime
    fastest_laps = valid_laps.loc[valid_laps.groupby('Team')['LapTime'].idxmin()]

    # Verifica se o SpeedST está sendo extraído corretamente
    if fastest_laps['SpeedST'].isnull().any():
        raise ValueError("Há valores nulos em 'SpeedST' nas voltas mais rápidas, verifique os dados.")

    # Organiza as voltas mais rápidas pela velocidade (SpeedST)
    max_speeds = fastest_laps.sort_values(by='SpeedST', ascending=True)

    # Coleta as informações necessárias
    top_speeds = max_speeds['SpeedST'].values
    team_names = max_speeds['Team'].values
    team_colors = [fastf1.plotting.get_team_color(lap.Team, session=session) for lap in max_speeds.itertuples()]

    # Criação do gráfico de barras horizontais
    bars = ax.barh(team_names, top_speeds, color=team_colors, edgecolor='grey')

    # Adicionar os rótulos de velocidade nas barras, corrigindo a exibição dos valores
    for bar, speed in zip(bars, top_speeds):
        ax.text(
            bar.get_width() + 2,  # Ajuste para garantir que o rótulo apareça após a barra
            bar.get_y() + bar.get_height() / 2,
            f'{speed:.2f} km/h',
            va='center',
            ha='left',
            fontsize=8,
            color='black',
            weight='bold'
        )

    # Adiciona o rodapé e ajusta o layout
    add_footer(ax)
    plt.tight_layout()

    # Salva o gráfico no buffer
    buf = io.BytesIO()
    plt.savefig(buf, format='png')
    buf.seek(0)
    plt.close()

    return buf

**Comment by theOehrly** [comment]
Are you talking about the speed trap values vs the car's telemetry data? If so, the speed traps are in a fixed location and this is not necessarily where the cars are the fastest on a straight. So the speed in the telemetry data can absolutely reach higher values.
